### DISCUSSION TITLE
Some summary, some ideas

### DISCUSSION THREAD

**Discussion by huang12zheng** [sean1832/GPT-Brain#16] [Ideas]
(1) Prepare various atomic contexts
(2) Find the context that most resembles the problem
(3) Give context and question to GPT

in(1), the context should be orthogonal, right?
Maybe (2) you can make a model to fine-tune it, but you're pretty straightforward

**Comment by sean1832** [comment]
Hi, I am not yet knowledgeable in the methodology of fine-tuning, but I am eager to learn. 😊
If you have experience with fine-tuning models, could you kindly provide some suggestions on how to approach fine-tuning and what objectives the fine-tuned model should aim for, if you feel comfortable doing so?
In (2), I am currently using embeddings to search for relevent context.

**Comment by huang12zheng** [comment]
https://www.bilibili.com/video/BV1wU4y1v7Eh/?p=75&spm_id_from=333.880.my_history.page.click&vd_source=6ce79c4a83aca838038629ac00dc610a

About finetune
I think just knowing Lora and Hard/Soft Prompt

Because I found a problem that CharGPT should just give you a word vector and doesn't give you all the features.
But maybe it's not a good choice because it can be complicated...

about (1)(3)
I want to give you something new, and I'm going to find that these two questions are very important.

The solution is clustering and hints
 
关于finetune
我觉得只要知道Lora和Hard/Soft Prompt

因为我发现一个问题, CharGPT应该只是给你一个词向量, 没有给你 所有的特征.
但是可能这不是一个好选择,因为这可能会比较复杂...

对于(1)(3)
我想给你一个新的东西,我会发现,这两个问题,很重要的.

解法是聚类和提示

[Chain-of-Thought Reasoning in Language Models 语言模型中的思维链推理](https://www.bilibili.com/video/BV12v4y1e7H7?spm_id_from=333.880.my_history.page.click)

**Comment by huang12zheng** [comment]
a new vedio
https://www.bilibili.com/video/BV1Xg411473w/?spm_id_from=333.337.search-card.all.click&vd_source=6ce79c4a83aca838038629ac00dc610a
you can just watch this instead of lihongyi's vedio
